lib/solveEquation/stepThrough.js

```diff
@@ -107,6 +157,69 @@ function stepThrough(leftNode, rightNode, comparator, debug=false) {
   return steps;
 }
 
+/*
+  Helper function that returns the roots and symbol of an input equation
+  that has roots to be found.
+  For every factor on the left hand side, solve a new equation that is factor = 0
+  for the symbol.
+  TODO: handle multiple variable solutions e.g (x + 2) (y + 2) = 0
+*/
+
+function getSolutionsAndSymbol (equation) {
+  const left = equation.leftNode;
+  const rightNode = Node.Creator.constant(0);
+
+  const solutions = [];
+
+  let symbol, steps, factorsWithSymbols;
+
+  // If left hand side is a power node and it does not resolve to a constant (e.g. 2^7)
+  // then it is a factor. If left hand side is a multiplication node, return a list of all the valid factors.
+  if (Node.Type.isOperator(left, '^') && !checks.resolvesToConstant(left)){
+    factorsWithSymbols = [left];
+  }
+  else {
+    factorsWithSymbols = equation.leftNode.args.filter(arg => !checks.resolvesToConstant(arg));
+  }
+
+  // For each factor, solve the equation factor = 0 for the symbol
+  // If the factor is a power node, solve the equation base = 0 for the symbol
```

Comment:
this stuff is pretty complex here but your comments are really helping - thanks!